### ISSUE TITLE
AllowCreate in NameIdPolicy is not correctly nullable

### ISSUE THREAD

**Issue by AtToftegaard** [ITfoxtec/ITfoxtec.Identity.Saml2#51]
Heya!

File _ITfoxtec.Identity.Saml2/src/ITfoxtec.Identity.Saml2/Util/GenericTypeConverter.cs_ contains this piece of code when converting the '**NameIdPolicy**'-Xml-element:
``` 
return GenericConvertValue<T, NameIdPolicy>(new NameIdPolicy
                {
                    AllowCreate = GenericConvertValue<bool, string>(xmlNode.Attributes[Schemas.Saml2Constants.Message.AllowCreate]?.Value),
                    Format = xmlNode.Attributes[Schemas.Saml2Constants.Message.Format]?.Value,
                    SPNameQualifier = xmlNode.Attributes[Schemas.Saml2Constants.Message.SpNameQualifier]?.Value
                });
```

We find that when a SAML-request contains the element 'NameIdPolicy' (triggering above code) but not 'AllowCreate', a type-cast exception is thrown:
```
System.InvalidCastException: Null object cannot be converted to a value type.
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at ITfoxtec.Identity.Saml2.Util.GenericTypeConverter.GenericConvertValue[T,U](U value)
   at ITfoxtec.Identity.Saml2.Util.GenericTypeConverter.ConvertElement[T](XmlNode xmlNode)
   at ITfoxtec.Identity.Saml2.XmlElementExtensions.GetElementOrNull[T](XmlElement xmlElement)
   at ITfoxtec.Identity.Saml2.Saml2AuthnRequest.Read(String xml, Boolean validateXmlSignature)
   at ITfoxtec.Identity.Saml2.Saml2RedirectBinding.Read(HttpRequest request, Saml2Request saml2RequestResponse, String messageName, Boolean validateXmlSignature)
   at ITfoxtec.Identity.Saml2.Saml2RedirectBinding.UnbindInternal(HttpRequest request, Saml2Request saml2RequestResponse, String messageName)
   at ITfoxtec.Identity.Saml2.Saml2Binding`1.Unbind(HttpRequest request, Saml2Request saml2Request)
```
We suspect `AllowCreate = GenericConvertValue<bool, string>` breaks if the bool is null.

Cheers.

**Comment by Revsgaard** [comment]
Thank you for the contribution.

I have added a [bug fix](https://github.com/ITfoxtec/ITfoxtec.Identity.Saml2/commit/bd51e8f950b34bc8241180cf43e7a65891263277) and released [version 4.0.6-beta2](https://github.com/ITfoxtec/ITfoxtec.Identity.Saml2/releases/tag/4.0.6-beta2). Please test and led me know if it solves your problem.

**Comment by AtToftegaard** [comment]
It does, thank you!
